### PR TITLE
Add backfill script

### PR DESCRIPTION
**Pull request by jaredsnyder**
Purpose of this PR is to add a script to do backfills.  I tried using `subprocess` but I couldn't get it to work.  `os.system` works just fine though.

### CODE REVIEW

File: backfill.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    """Return list of months between start_month and end_month inclusive.
+
+    Parameters
+    ----------
+    start_month : str
+        Earliest month to include in the backfill
+        Must have format %Y-%m
+    end_month : str
+        Final month to include in the backfill
+        Must have format %Y-%m
```
Comment (m-d-bowerman) [inline backfill.py:26]: Maybe too philosophical, but is this better than accepting any date string and truncating to the 1st of the month?

File: backfill.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        Indicates whether or not to use the test or prod table
+
+    Returns
+    -------
+    str
+        Full output table
+    """
+    with open(config_path, "rb") as infile:
+        config_data = yaml.safe_load(infile)
+    if test_mode:
```
Comment (m-d-bowerman) [inline backfill.py:63]: Probably a good idea to add details about the config formats in the README, especially the prod and test dataset distinction and that test set is required to run in test mode.

File: backfill.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        # case where testing locally
+        output_info = config_data["output"]["test"]
+    else:
+        output_info = config_data["output"]["prod"]
+    target_table = (
+        f"{output_info['project']}.{output_info['database']}.{output_info['table']}"
```
Comment (m-d-bowerman) [inline backfill.py:69]: Sorry for the nit, but it should be `'dataset'`, not `'database'`. Might be confusing

File: backfill.py (hunk 1 above)
Comment (jaredsnyder) [reply backfill.py:26]: I think so, just because I don't want ambiguity about how these are monthly forecasts

File: backfill.py (hunk 3 above)
Comment (jaredsnyder) [reply backfill.py]: fixed

**Pull request merged**